src/repositories/request-repository.ts

```diff
@@ -0,0 +1,80 @@
+import CID from "cids";
+import { EntityRepository, InsertResult, UpdateResult } from "typeorm";
+import { BaseRepository } from 'typeorm-transactional-cls-hooked';
+
+import Context from "../context";
+import Contextual from "../contextual";
+import { Request, RequestUpdateFields } from "../models/request";
+import { RequestStatus } from "../models/request-status";
+import { config } from "node-config-ts";
+
+@EntityRepository(Request)
+export default class RequestRepository extends BaseRepository<Request> implements Contextual {
+
+  setContext(context: Context): void {}
+
+  /**
+   * Create/updates client request
+   * @param request - Request
+   */
+  public async createOrUpdate(request: Request): Promise<Request> {
+    return this.manager.getRepository(Request).save(request);
+  }
+
+  /**
+   * Creates client requests
+   * @param requests - Requests
+   */
+  public async createRequests(requests: Array<Request>): Promise<InsertResult> {
+    return this.manager.getRepository(Request)
+      .createQueryBuilder()
+      .insert()
+      .into(Request)
+      .values(requests)
+      .execute();
+  }
+
+  /**
+   * Create/updates client requests
+   * @param ids - Request IDs
+   * @param fields - Fields to update
+   */
+  public async updateRequests(fields: RequestUpdateFields, ids: number[]): Promise<UpdateResult> {
+    return this.manager.getRepository(Request)
+      .createQueryBuilder()
+      .update(Request)
+      .set(fields)
+      .whereInIds(ids)
+      .execute();
+  }
+
+  /**
+   * Gets all requests by status
```

Comment:
How about something like:
"Gets all requests in need of processing.  Requests that need processing are Requests that have been received but have never had anchor requests submitted for them, as well as Requests that have been waiting to be successfully anchored for longer than 'expirationPeriod'"